File: Lesson6_Maskaikin/app/src/main/java/com/vlad/lesson6_maskaikin/Fragment3.java
package com.vlad.lesson6_maskaikin;


import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


public class Fragment3 extends Fragment {

    public final static String TAG_VIEW_PAGER = "view pager";

    public Fragment3() {

    }

    public static Fragment3 getInstance() {
        Fragment3 fragment3 = new Fragment3();
        return fragment3;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment

        View layoutView = inflater.inflate(R.layout.fragment_3, container, false);
        final TextView textViewFragment3 = layoutView.findViewById(R.id.textViewFragment3);

        textViewFragment3.setOnClickListener(new View.OnClickListener() {
            Boolean checkClick = false;

            @Override
            public void onClick(View view) {
                if (!checkClick) {
                    getChildFragmentManager()
                            .beginTransaction()
                            .add(R.id.containerInFragment3, Fragment3ViewPager.getInstance(), TAG_VIEW_PAGER)
                            .commit();
                    textViewFragment3.setText(R.string.collapse_banner);
                    checkClick = true;
                } else {
                    Fragment fragment = getChildFragmentManager().findFragmentByTag(TAG_VIEW_PAGER);
                    if (fragment != null) {
                        getChildFragmentManager().beginTransaction().remove(fragment).commit();
                    }
                    textViewFragment3.setText(R.string.show_banner);
                    checkClick = false;
                }
            }
        });
        return layoutView;
    }


}

File: Lesson6_Maskaikin_Kotlin/app/src/main/java/com/vlad/lesson6_maskaikin_kotlin/Fragment3.kt
package com.vlad.lesson6_maskaikin_kotlin


import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView


class Fragment3 : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        val layoutView = inflater.inflate(R.layout.fragment_3, container, false)
        val textViewFragment3 = layoutView.findViewById<TextView>(R.id.textViewFragment3)

        textViewFragment3.setOnClickListener(object : View.OnClickListener {
            var checkClick: Boolean = false

            override fun onClick(view: View) {
                if (!checkClick) {
                    childFragmentManager
                            .beginTransaction()
                            .add(R.id.containerInFragment3, Fragment3ViewPager.getInstance(), TAG_VIEW_PAGER)
                            .commit()
                    textViewFragment3.setText(R.string.collapse_banner)
                    checkClick = true
                } else {
                    val fragment = childFragmentManager.findFragmentByTag(TAG_VIEW_PAGER)
                    if (fragment != null) {
                        childFragmentManager.beginTransaction().remove(fragment).commit()
                    }
                    textViewFragment3.setText(R.string.show_banner)
                    checkClick = false
                }
            }
        })
        return layoutView
    }

    companion object {

        const val TAG_VIEW_PAGER = "view pager"


        fun getInstance() : Fragment3 {
            val fragment3 = Fragment3()
            return fragment3
        }
    }


}
